conda_forge_tick/migrators/staticlib.py

```diff
@@ -0,0 +1,581 @@
+import logging
+import os
+import re
+import typing
+from functools import lru_cache
+from typing import Any, Optional, Sequence
+
+import networkx as nx
+import rapidjson
+from conda.models.match_spec import MatchSpec
+from conda.models.records import PackageRecord
+from conda.models.version import VersionOrder
+from conda_forge_metadata.repodata import fetch_repodata
+
+from conda_forge_tick.contexts import ClonedFeedstockContext, FeedstockContext
+from conda_forge_tick.migrators.core import (
+    GraphMigrator,
+    MiniMigrator,
+    _gen_active_feedstocks_payloads,
+)
+from conda_forge_tick.os_utils import pushd
+from conda_forge_tick.utils import (
+    extract_section_from_yaml_text,
+    get_keys_default,
+    get_migrator_name,
+)
+
+if typing.TYPE_CHECKING:
+    from ..migrators_types import AttrsTypedDict, MigrationUidTypedDict
+
+BUILD_STRING_END_RE = re.compile(r".*_\d+$")
+
+logger = logging.getLogger(__name__)
+
+
+def _left_gt_right_rec(lrec, rrec):
+    lver = VersionOrder(lrec.version)
+    lbuild = lrec.build_number
+    rver = VersionOrder(rrec.version)
+    rbuild = rrec.build_number
+
+    if (lver > rver) or (lver == rver and lbuild > rbuild):
+        return True
+    else:
+        return False
+
+
+@lru_cache(maxsize=128)
+def _cached_match_spec(req: str) -> MatchSpec:
+    return MatchSpec(req)
+
+
+@lru_cache(maxsize=128)
+def _match_spec_is_exact(ms: MatchSpec) -> bool:
```

Comment:
There's a `MatchSpec.strictness` attribute for this. Max value being 3 which is what "exact" means here.

Reply:
Nope that doesn't work:

```python
In [70]: MatchSpec("libfoo 10 *_5").strictness
Out[70]: 3

In [71]: MatchSpec("libfoo 10 h4543_5").strictness
Out[71]: 3

In [72]: MatchSpec("libfoo 10 *_5").get_exact_value("build")

In [73]: MatchSpec("libfoo 10 h4543_5").get_exact_value("build")
Out[73]: 'h4543_5'
```